### PR TITLE
Fix opentelemetry-proto file exists check

### LINKED ISSUE #1823: opentelemetry-proto local own dependency not found
**Describe your environment**

* Ubuntu 20.04 cmake version `3.16.3`  
* RHEL 8.5 cmake version `3.20.2-4.el8`
* Ubuntu 20.04 cmake version `3.25.0` (latest as of today) 

Tested with opentelemetry-cpp [latest](https://github.com/open-telemetry/opentelemetry-cpp/commit/7e26a7e84e94a50a0b04e70ed88ac04d75afbf40) (as of today) and opentelemetry-proto [v0.19.0](https://github.com/open-telemetry/opentelemetry-proto/tree/v0.19.0)

**Steps to reproduce**
Building with `OTELCPP_PROTO_PATH` set

```
cmake -DOTELCPP_PROTO_PATH=/path/to/proto ..
```

**What is the expected behavior?**
local opentelemetry dependency is found and used to build opentelemetry-cpp

**What is the actual behavior?**
it always fails with "not existing" error

```
CMake Error at cmake/opentelemetry-proto.cmake:30 (message):
  OTELCPP_PROTO_PATH does not point to a opentelemetry-proto repository
Call Stack (most recent call first):
  CMakeLists.txt:328 (include)
```

**Additional context**

### PR DESCRIPTION
**Pull request by eguzki**
Fixes #1823

## Changes

Fix CMake `EXISTS` syntax

For significant contributions please make sure you have completed the following items:

* [x] `CHANGELOG.md` updated for non-trivial changes
* [ ] Unit tests have been added
* [ ] Changes in public API reviewed

### CODE REVIEW

**Review by marcalff — APPROVED** [review]
Thanks for the fix.

The original bug is mine, sorry I must have been distracted when doing last minute changes.

**Review by esigo — APPROVED** [review]

File: cmake/opentelemetry-proto.cmake [hunk 1]
```diff
@@ -26,7 +26,7 @@
 #
 
 if(OTELCPP_PROTO_PATH)
-  if(NOT EXISTS(${OTELCPP_PROTO_PATH}/opentelemetry/proto/common/v1/common.proto))
```
Comment (ThomsonTan) [inline cmake/opentelemetry-proto.cmake:29]: @marcalff , do you think we can have a test on this passed in env var?

Comment (marcalff) [reply cmake/opentelemetry-proto.cmake:29]: Tested locally with:

```
malff@localhost.localdomain:build-opentelemetry-cpp> pwd
/home/malff/CODE/MY_GITHUB/build-opentelemetry-cpp
```

```
ccmake \
-DOTELCPP_PROTO_PATH=/home/malff/CODE/MY_GITHUB/opentelemetry-proto \
-DWITH_OTLP=ON \
-DWITH_OTLP_HTTP=ON \
-DWITH_OTLP_GRPC=ON \
 ../opentelemetry-cpp \
```

CMake result without the fix:

```
 CMake Error at cmake/opentelemetry-proto.cmake:30 (message):
   OTELCPP_PROTO_PATH does not point to a opentelemetry-proto repository
```

CMake result with the fix:

```
 opentelemetry-proto dependency satisfied by: external path
```

So the fix is working.

OTELCPP_PROTO_PATH also shows in the ccmake UI.

@ThomsonTan 
Using environment variables will be odd, may be I don't understand what you suggest.

Comment (lalitb) [reply cmake/opentelemetry-proto.cmake:29]: @ThomsonTan  - Hope this is ok to merge. If there are any specific test scenarios, we can address them separately.

**Pull request merged**